contracts/service-registry/src/state.rs

```diff
@@ -184,6 +184,27 @@ pub fn register_chains_support(
     Ok(())
 }
 
+pub fn deregister_chains_support(
+    storage: &mut dyn Storage,
+    service_name: String,
+    chains: Vec<ChainName>,
+    worker: WorkerAddress,
+) -> Result<(), ContractError> {
+    CHAINS_PER_WORKER.update(storage, (&service_name, &worker), |current_chains| {
```

Comment:
This function allows workers to deregister for chains that they are not registered for. Is that intended? One side effect is that if a worker is not registered for any chains, this function is going to write an empty hashset into CHAINS_PER_WORKER

Reply:
yeah it would not be an error.